Zend/Optimizer/zend_dump.c

```diff
@@ -67,10 +67,8 @@ void zend_dump_const(const zval *zv)
 			fprintf(stderr, " float(%g)", Z_DVAL_P(zv));
 			break;
 		case IS_STRING:;
-			zend_string *escaped_string = php_addcslashes(Z_STR_P(zv), "\"\\", 2);
-
-			fprintf(stderr, " string(\"%s\")", ZSTR_VAL(escaped_string));
-
+			zend_string *escaped_string = php_repr_str(Z_STR_P(zv)->val, Z_STR_P(zv)->len);
+			fprintf(stderr, " string(%s)", ZSTR_VAL(escaped_string));
```

Comment:
We can achieve a similar result with `php_addcslashes(Z_STR_P(zv), "\x00..\x1f\\\\\x7f..\xff", 2)` (with the different that it uses an octal representation), or the same result with `smart_str_append_escaped()`.